src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -67,7 +67,8 @@ public void testDoPost_NewUser() throws IOException, ServletException {
 
     Mockito.verify(mockUserStore).addUser(userArgumentCaptor.capture());
     Assert.assertEquals(userArgumentCaptor.getValue().getName(), "test username");
-    Assert.assertEquals(userArgumentCaptor.getValue().getPassword(), "test password");
+    Assert.assertThat(
+        userArgumentCaptor.getValue().getPasswordHash(), CoreMatchers.containsString("$2a$10$"));
```

Comment:
Perhaps you should also do a test based on expected length. That will protect against possible truncation or other similar corruptions.

Reply:
That is, if the length is fixed and known.